Sources/Build/BuildPlan.swift

```diff
@@ -1603,9 +1596,10 @@ public class BuildPlan {
             }
         }
 
+        // add test manifest target on non-darwin platforms
         if !buildParameters.triple.isDarwin() {
-            if product.type == .test {
```

Comment:
👀 note that this still checks for `buildParameters.triple.isDarwin`, which I proposed we also change such that SwiftPM *always* uses test discovery on all platforms and not rely on XCTest on Darwin. I can do this in follow up PR or change this PR if folks agree this is a good change.

Reply:
@abertelrud @neonichu ^^

Reply:
I think follow-up PR would be better and we might want to do a forum discussion on that topic specifically.

Reply:
I agree it can be done in a follow-up PR, but my general view on this is that `swift test` should have consistent behavior across all platforms, as much as possible.  So that would argue for using test discovery everywhere.  Since not everyone can easily test for all platforms locally, minimizing the differences makes it less likely that a change will fail on some platforms but succeed on others.  But can be done in a follow-on PR.